### PR TITLE
Setup classes associations

### PR DESCRIPTION
**Pull request by abiola-arobieke**
In this milestone:

The book.rb, classroom.rb, and rental.rb files were added to the project.
Has-many/belongs-to relationship was performed between the class Classroom and Student.
The many-to-many relationship was performed between class Person and Book using the intermediate class Rental.

### CODE REVIEW

File: classes/rental.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Define class rentals
+class Rental
+  attr_reader :date
```
Comment (ishimwezachee) [inline classes/rental.rb]: - [ ]  Good job for creating the attributes for Rental class. One thing to be improved here is that date attribute should have both `setter `and `getter`.  When you use `attr_reader `on date, you limit it to be changed or updated . fix it for your app to work properly.

File: classes/book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Define class Book
+class Book
+  attr_reader :rentals
```
Comment (ishimwezachee) [inline classes/book.rb]: - [x]  Good job for creating the attributes for Book class. One thing to be improved here is that Rental attribute should have both `setter `and `getter`.  When you use `attr_reader `on Rental , you limit it to be changed or updated . fix it for your app to work properly.

File: classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (ishimwezachee) [inline classes/book.rb:9]: - [x]  Great job for creating Book class, but seemed that you might have forgetted to add `add_rental method`. This method should present in your book class as well to help you to add both person and date when the book is rented, for more [check ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`)

File: classes/person.rb [hunk 4]
```diff
@@ -4,14 +4,15 @@
...
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (ishimwezachee) [inline classes/person.rb:16]: - [ ]  Great job for creating Person class, but seemed that you might have forgotten to add `add_rental method`. This method should present in your Person class as well to help you to add both book and date when a person rented, for more [check ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`)

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @abiola-arobieke , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image

## Highlights
✔ Clean and well organized codes 
✔ All classes were implemented 
✔ relationship has been created between classes 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/person.rb [hunk 5]
```diff
@@ -28,4 +29,8 @@ def can_use_services
   def correct_name
     @name
   end
+
+  def add_rental(person)
+    @rentals.push(person)
+  end
```
Comment (Deepakdanger) [inline classes/person.rb]: - [ ] Here we have to create rental so kindly add 
 ```
def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

File: classes/book.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+  def add_book_rental(book)
+    @rentals.push(book)
+  end
```
Comment (Deepakdanger) [inline classes/book.rb]: - [ ] Here also we have to create a rental, so kindly add code to create rental here.
```ruby
 def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @abiola-arobieke , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @abiola-arobieke 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**